test/test_pipeline/components/data_preprocessing/test_data_preprocessing.py

```diff
@@ -23,19 +29,60 @@ def do_a_fit_transform(self, sparse_output):
             [1, 2, 0],
             [3, 0, 0],
             [2, 9, np.nan]])
+        # After the preprocessing of X_cat, we expect that:
+        # The missing value in feature 3 gets imputed as category on its own.
+        # Features get encoded by one hot encoding.
+        # Therefore, Y_cat is what we should get after data preprocessing X_cat:
         Y_cat = np.array([
             [1, 0, 0, 0, 1, 0, 0, 1],
             [0, 0, 1, 1, 0, 0, 0, 1],
             [0, 1, 0, 0, 0, 1, 1, 0]])
-        # Combined dataset with shuffled columns:
+        # Combine datasets and shuffle columns:
         X_comb = np.hstack((X_num, X_cat))
         categ_feat = np.array([False] * 3 + [True] * 3)
         random_order = np.random.choice(np.arange(6), size=6, replace=False)
         X_comb = X_comb[:, random_order]
         categ_feat = categ_feat[random_order]
-        # fit_transform
-        Y_comb = DataPreprocessor(
-            categorical_features=categ_feat, sparse=sparse_output).fit_transform(X_comb)
+        # Data preprocessing
+        DPP = DataPreprocessor(categorical_features=categ_feat, sparse=sparse_output)
+        Y_comb = DPP.fit_transform(X_comb)
+        Y_comb = Y_comb.todense() if sparse_output else Y_comb
+        # check shape
+        self.assertEquals(Y_comb.shape, (3, 10))
+        # check content. Categorical columns appear first in Y.
+        for row in range(3):
+            self.assertAlmostEqual(np.sum(Y_cat[row]), np.sum(Y_comb[row, :8]))
+            self.assertAlmostEqual(np.sum(Y_num[row]), np.sum(Y_comb[row, 8:]))
+        #
+        # Now we do everything again, but using a different dataset and the already
+        # fitted data preprocessor
+        #
+
+        # Numerical dataset
+        X_num = np.array([
+            [1., 5.,     1.],   # noqa : matrix legibility
+            [2., np.nan, 0.],   # noqa : matrix legibility
+            [3., 1.,     3.]])  # noqa : matrix legibility
+        # Y_num is what we should get after data preprocessing X_num
+        Y_num = np.array([
+            [ 3/sdev, -1/sdev],   # noqa : matrix legibility
+            [     0., -2/sdev],   # noqa : matrix legibility
+            [-1/sdev,  1/sdev]])  # noqa : matrix legibility
+        # Categorical dataset
+        X_cat = np.array([
+            [1, 2,      1],
+            [3, 0,      np.nan],
+            [2, np.nan, 0]])
+        # Y_cat is what we should get after data preprocessing X_cat
+        Y_cat = np.array([
+            [1, 0, 0, 0, 1, 0, 0, 0],
+            [0, 0, 1, 1, 0, 0, 1, 0],
+            [0, 1, 0, 0, 0, 0, 0, 1]])
+        # Combine datasetd and shuffle columns:
+        X_comb = np.hstack((X_num, X_cat))
+        X_comb = X_comb[:, random_order]
+        # Data preprocessing with already fitted transformer
+        Y_comb = DPP.transform(X_comb)
         Y_comb = Y_comb.todense() if sparse_output else Y_comb
         # check shape
         self.assertEquals(Y_comb.shape, (3, 10))
```

Comment:
Would it be possible that you check for the actual output? (and above, too?

Reply:
Done.